### PR TITLE
Fix parsing `...` in arguments

### LINKED ISSUE #1830: Prism accepts argument after forwarding arguments like `f(..., 1)`
```ruby
# Should be syntax error
Prism.parse('def f(...); p(..., 1); end').success? #=> true
Prism.parse('def f(...); p(..., x: 1); end').success? #=> true
Prism.parse('def f(...); p(..., &b); end').success? #=> true
Prism.parse('def f(...); p(..., ...); end').success? #=> true
Prism.parse('def f(...); p(...){}; end').success? #=> true

# syntax error, unexpected ',', expecting ')' (SyntaxError)
eval 'def f(...); p(..., 1); end'
eval 'def f(...); p(..., x: 1); end'
eval 'def f(...); p(..., &b); end'
eval 'def f(...); p(..., ...); end'

# => both block arg and actual block given (SyntaxError)
eval 'def f(...); p(...){}; end'
```

### LINKED ISSUE #1831: Prism accepts forward arguments without parenthesis `f 1, ...`
```ruby
Prism.parse('def f(...); p 1, ...; end').success? #=> true
eval('def f(...); p 1, ...; end') #=> syntax error, unexpected ';' (SyntaxError)
```

### PR DESCRIPTION
**Pull request by makenowjust**
Fix #1830
Fix #1831

Both fixes of #1830 and #1831 are small and modify near lines, so I fixed them in one commit.

### CODE REVIEW

File: include/prism/diagnostic.h [hunk 1]
```diff
@@ -40,12 +40,14 @@ typedef enum {
...
     PM_ERR_AMPAMPEQ_MULTI_ASSIGN,
     PM_ERR_ARGUMENT_AFTER_BLOCK,
+    PM_ERR_ARGUMENT_AFTER_FORWARDING_ELLIPSES,
     PM_ERR_ARGUMENT_BARE_HASH,
     PM_ERR_ARGUMENT_BLOCK_MULTI,
     PM_ERR_ARGUMENT_FORMAL_CLASS,
     PM_ERR_ARGUMENT_FORMAL_CONSTANT,
     PM_ERR_ARGUMENT_FORMAL_GLOBAL,
     PM_ERR_ARGUMENT_FORMAL_IVAR,
+    PM_ERR_ARGUMENT_FORWARDING_ELLIPSIS_IN_NO_PAREN,
```
Comment (makenowjust) [inline include/prism/diagnostic.h:50]: This constant name seems too long. Can you suggest a better name?

Comment (kddnewton) [reply include/prism/diagnostic.h:50]: `PM_ERR_ARGUMENT_FORWARDING_UNBOUND`?

**Review by kddnewton — APPROVED** [review]
One small naming suggestion.

File: src/diagnostic.c [hunk 2]
```diff
@@ -51,210 +51,210 @@
...
+    [PM_ERR_AMPAMPEQ_MULTI_ASSIGN]              = "Unexpected `&&=` in a multiple assignment",
+    [PM_ERR_ARGUMENT_AFTER_BLOCK]               = "Unexpected argument after a block argument",
+    [PM_ERR_ARGUMENT_AFTER_FORWARDING_ELLIPSES] = "Unexpected argument after `...`",
+    [PM_ERR_ARGUMENT_BARE_HASH]                 = "Unexpected bare hash argument",
+    [PM_ERR_ARGUMENT_BLOCK_MULTI]               = "Multiple block arguments; only one block is allowed",
+    [PM_ERR_ARGUMENT_FORMAL_CLASS]              = "Invalid formal argument; formal argument cannot be a class variable",
+    [PM_ERR_ARGUMENT_FORMAL_CONSTANT]           = "Invalid formal argument; formal argument cannot be a constant",
+    [PM_ERR_ARGUMENT_FORMAL_GLOBAL]             = "Invalid formal argument; formal argument cannot be a global variable",
+    [PM_ERR_ARGUMENT_FORMAL_IVAR]               = "Invalid formal argument; formal argument cannot be an instance variable",
+    [PM_ERR_ARGUMENT_FORWARDING_UNBOUND]        = "Unexpected `...` in an non-parenthesized call",
```
Comment (makenowjust) [inline src/diagnostic.c:64]: @kddnewton Does this error seem good to you?

Comment (kddnewton) [reply src/diagnostic.c:64]: Looks good!

**Pull request merged**